mission/mission_handler_hold_position.hxx

```diff
@@ -0,0 +1,165 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+
+/*******************************************************************************
+ * \file mission_handler_hold_position.hxx
+ *
+ * \author MAV'RIC Team
+ * \author Matthew Douglas
+ *
+ * \brief The MAVLink mission planner handler functions for the hold position state
+ *
+ ******************************************************************************/
+
+
+#ifndef MISSION_HANDLER_HOLD_POSITION_HXX__
+#define MISSION_HANDLER_HOLD_POSITION_HXX__
+
+template <class T>
+Mission_handler_hold_position<T>::Mission_handler_hold_position(T& controller,
+                                                                const INS& ins):
+            Mission_handler(),
+            controller_(controller),
+            ins_(ins)
+{
+    waypoint_ = Waypoint(   MAV_FRAME_LOCAL_NED,
+                            MAV_CMD_NAV_LOITER_UNLIM,
+                            0,
+                            0.0f,
+                            0.0f,
+                            0.0f,
+                            0.0f,
+                            0.0f,
+                            0.0f,
+                            0.0f);
+}
+
+template <class T>
+bool Mission_handler_hold_position<T>::can_handle(const Waypoint& wpt) const
+{
+    bool handleable = false;
+
+    uint16_t cmd = wpt.command();
+    if (cmd == MAV_CMD_NAV_LOITER_UNLIM ||
+        cmd == MAV_CMD_NAV_LOITER_TIME  ||
+        cmd == MAV_CMD_NAV_LOITER_TO_ALT ||
+        (cmd == MAV_CMD_OVERRIDE_GOTO && wpt.param1() == MAV_GOTO_DO_HOLD))
+    {
+        handleable = true;
+    }
+
+    return handleable;
+}
+
+template <class T>
+bool Mission_handler_hold_position<T>::setup(const Waypoint& wpt)
+{
+    bool success = true;
+    waypoint_ = wpt;
+    start_time_ = time_keeper_get_ms();
+    within_radius_ = false;
+
+    return success;
+}
+
+template <class T>
+Mission_handler::update_status_t Mission_handler_hold_position<T>::update()
+{
+    // Set goal
+    bool ret = set_control_command();
+
+    /*******************
+    Determine status code 
+    ********************/
+    // Determine if we have entered the hold position volume
+    local_position_t wpt_pos = waypoint_.local_pos();
+
+    float radius;
+    if (!waypoint_.radius(radius))
+    {
+        radius = 0.0f;
```

Comment:
This is useless since waypoint.radius() is already setting the radius to zero in case no radius correspond to that type of waypoint